### PR TITLE
Allow to override binding of Hdfs filesystem for caching

### PR DESCRIPTION
**Pull request by sopel39**

### CODE REVIEW

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java [hunk 1]
```diff
@@ -26,6 +26,6 @@
 @Retention(RUNTIME)
 @Target({FIELD, PARAMETER, METHOD})
 @Qualifier
-public @interface ForRubix
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java:29]: I don't see a reason for rename. I find it better the way it was.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixModule.java [hunk 2]
```diff
@@ -22,6 +22,7 @@
 import com.qubole.rubix.prestosql.CachingPrestoDistributedFileSystem;
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixModule.java:22]: Update commit message to:
```
Do not allow impersonation with DefaultRubixHdfsInitializer
```

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java (hunk 1 above)
Comment (sopel39) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java:29]: If the name is the same, it wouldn't be obvious what's the binding for. This is specifically for binding `Hdfs` file system. Other FS are still statically bound in `io.prestosql.plugin.hive.rubix.RubixInitializer#updateRubixConfiguration`

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixModule.java (hunk 2 above)
Comment (sopel39) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixModule.java:22]: The check was actually moved from `io.prestosql.plugin.hive.authentication.HiveAuthenticationModule#setup`. updated code

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java (hunk 1 above)
Comment (kokosing) [reply presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java:29]: > If the name is the same, it wouldn't be obvious what's the binding for

That is correct, but this is why binding requires the class and annotation. Class is the most important part while annotation is just auxiliary.

File: presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java [hunk 3]
```diff
@@ -13,19 +13,9 @@
...
-import static java.lang.annotation.ElementType.FIELD;
-import static java.lang.annotation.ElementType.METHOD;
-import static java.lang.annotation.ElementType.PARAMETER;
-import static java.lang.annotation.RetentionPolicy.RUNTIME;
-
-@Retention(RUNTIME)
-@Target({FIELD, PARAMETER, METHOD})
-@Qualifier
-public @interface ForRubix
+public interface RubixHdfsInitializer
```
Comment (kokosing) [inline presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/RubixHdfsInitializer.java:18]: One another idea is to use guice multibinder where you could have several independent `ConfigurationInitializer`.

**Pull request merged**